### PR TITLE
[14.0][MIG] queue job: Migration to 14.0

### PR DESCRIPTION
**Pull request by eantones**

### CODE REVIEW

**Review by guewen — APPROVED** [review]
Thanks! (reviewed migration commits, untested)

**Review by bealdav — APPROVED** [review]
The last commit implies very small diff with previous version

Selection_106

Just new required changes in v14.

LGTM

**Review by atchuthan — APPROVED** [review]

**Comment by guewen** [comment]
I'd like to take the chance of removing completely the decorators in 14.0 (started for 13.0 #274):

* 12.0: deprecate decorators but show no warning when they are used
* 13.0: deprecate decorators and show warnings when they are used
* 14.0: entirely remove them, as we are starting migrations, it will be easy to tackle in the modules

**Comment by guewen** [comment]
Thanks for the migration BTW :)

**Comment by atchuthan** [comment]
@eantones  
When I tried to test this PR for queue_job in my local DB, I got a few strange warnings in the odoo log.

```
WARNING testv14 odoo.fields: Field queue.job.record_ids: unknown parameter '_base_type', if this is an actual parameter you may want to override the method _valid_field_parameter on the relevant model in order to allow it
WARNING testv14 odoo.fields: Field queue.job.args: unknown parameter '_base_type', if this is an actual parameter you may want to override the method _valid_field_parameter on the relevant model in order to allow it
WARNING testv14 odoo.fields: Field queue.job.kwargs: unknown parameter '_base_type', if this is an actual parameter you may want to override the method _valid_field_parameter on the relevant model in order to allow it
```

**Comment by atchuthan** [comment]
Same warning in runbot log too [https://runbot2-3.odoo-community.org/runbot/static/build/3448412-263-8a6f92/logs/job_20_test_all.txt](url)

2020-11-04 06:41:14,707 159 INFO openerp_test odoo.modules.loading: updating modules list
2020-11-04 06:41:14,709 159 INFO openerp_test odoo.addons.base.models.ir_module: ALLOW access to module.update_list on [] to user __system__ #1 via n/a
2020-11-04 06:41:15,486 159 INFO openerp_test odoo.addons.base.models.ir_module: ALLOW access to module.button_install on ['queue_job'] to user __system__ #1 via n/a
2020-11-04 06:41:15,590 159 INFO openerp_test odoo.modules.loading: loading 22 modules...
2020-11-04 06:41:15,860 159 INFO openerp_test odoo.modules.loading: 22 modules loaded in 0.27s, 0 queries (+0 extra)
2020-11-04 06:41:15,863 159 INFO openerp_test odoo.modules.loading: loading 23 modules...
2020-11-04 06:41:15,863 159 INFO openerp_test odoo.modules.loading: Loading module queue_job (20/23)
2020-11-04 06:41:16,500 159 WARNING openerp_test odoo.fields: Field queue.job.record_ids: unknown parameter '_base_type', if this is an actual parameter you may want to override the method _valid_field_parameter on the relevant model in order to allow it
2020-11-04 06:41:16,500 159 WARNING openerp_test odoo.fields: Field queue.job.args: unknown parameter '_base_type', if this is an actual parameter you may want to override the method _valid_field_parameter on the relevant model in order to allow it
2020-11-04 06:41:16,500 159 WARNING openerp_test odoo.fields: Field queue.job.kwargs: unknown parameter '_base_type', if this is an actual parameter you may want to override the method _valid_field_parameter on the relevant model in order to allow it
2020-11-04 06:41:16,542 159 INFO openerp_test odoo.modules.registry: module queue_job: creating or updating database tables
2020-11-04 06:41:17,529 159 INFO openerp_test odoo.modules.loading: loading queue_job/security/security.xml
2020-11-04 06:41:17,626 159 INFO openerp_test odoo.modules.loading: loading queue_job/security/ir.model.access.csv
2020-11-04 06:41:17,664 159 INFO openerp_test odoo.modules.loading: loading queue_job/views/queue_job_views.xml
2020-11-04 06:41:17,839 159 INFO openerp_test odoo.modules.loading: loading queue_job/data/queue_data.xml

File: queue_job/models/queue_job.py [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+    def _message_post_on_failure(self):
+        # subscribe the users now to avoid to subscribe them
+        # at every job creation
+        domain = self._subscribe_users_domain()
+        users = self.env["res.users"].search(domain)
+        self.message_subscribe(partner_ids=users.mapped("partner_id").ids)
+        for record in self:
+            msg = record._message_failed_job()
+            if msg:
+                record.message_post(body=msg, subtype="queue_job.mt_job_failed")
```
Comment (CasVissers-360ERP) [inline queue_job/models/queue_job.py]: Should be changed to subtype_xmlid

**Review by CasVissers-360ERP — CHANGES_REQUESTED** [review]
Got a traceback for wrong subtype usage while testing.

**Comment by CasVissers-360ERP** [comment]
Additional:
Field queue.job.record_ids: unknown parameter '_base_type', if this is an actual parameter you may want to override the method _valid_field_parameter on the relevant model in order to allow it

**Comment by sebastienbeau** [comment]
Hi, I have rebase the work done here : https://github.com/OCA/queue/pull/286
For queue job we avoid starting from a blank repo as most of the module are always migrated.
So we prefers to keep a common history between branch (easier for maintaining)

**Pull request closed without merging**